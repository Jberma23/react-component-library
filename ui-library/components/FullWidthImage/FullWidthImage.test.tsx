import { render } from "@testing-library/react";
import { FullWidthImage } from ".";

describe(FullWidthImage, () => {
  it("renders with defaults", () => {
    const { asFragment } = render(
      <FullWidthImage
        src="http://www.not-a-real-url-this-is-just-for-testing.com/image.png"
        alt="This is a test image"
        captionText="This is a test image caption"
        aspectRatio="16/9"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders without a caption", () => {
    const { asFragment } = render(
      <FullWidthImage
        src="http://www.not-a-real-url-this-is-just-for-testing.com/image.png"
        alt="This is a test image"
        aspectRatio="1/1"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders without alt text", () => {
    const { asFragment } = render(
      <FullWidthImage
        src="http://www.not-a-real-url-this-is-just-for-testing.com/image.png"
        captionText="This is a test image caption"
        aspectRatio="4/3"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("uses its height/width as the aspect ratio when it is passed", () => {
    render(
      <FullWidthImage
        src="http://www.not-a-real-url-this-is-just-for-testing.com/image.png"
        captionText="This is a test image caption"
        height={1999}
        width={2000}
      />
    );

    expect(
      document.querySelector<HTMLElement>("figure > div")?.style
    ).toMatchObject({
      aspectRatio: "2000/1999",
    });
  });

  it("passes all a11y checks", async () => {
    await expect(
      <FullWidthImage
        src="http://www.not-a-real-url-this-is-just-for-testing.com/image.png"
        alt="This is a test image"
        captionText="This is a test image caption"
        aspectRatio="5/4"
      />
    ).toBeA11yWhenRendered();
  });
});
