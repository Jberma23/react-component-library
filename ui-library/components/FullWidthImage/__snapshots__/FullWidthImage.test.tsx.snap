// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullWidthImage renders with defaults 1`] = `
<DocumentFragment>
  <figure>
    <div
      class="relative overflow-hidden mb-xs aspect-16/9"
    >
      <img
        alt="This is a test image"
        class="object-cover"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=3840&q=100"
        srcset="/_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=640&q=100 640w, /_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=750&q=100 750w, /_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=828&q=100 828w, /_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=1080&q=100 1080w, /_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=1200&q=100 1200w, /_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=1920&q=100 1920w, /_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=2048&q=100 2048w, /_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=3840&q=100 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
    </div>
    <figcaption
      class="text-body-small text-gray-700"
    >
      This is a test image caption
    </figcaption>
  </figure>
</DocumentFragment>
`;

exports[`FullWidthImage renders without a caption 1`] = `
<DocumentFragment>
  <figure>
    <div
      class="relative overflow-hidden mb-xs aspect-1/1"
    >
      <img
        alt="This is a test image"
        class="object-cover"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=3840&q=100"
        srcset="/_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=640&q=100 640w, /_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=750&q=100 750w, /_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=828&q=100 828w, /_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=1080&q=100 1080w, /_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=1200&q=100 1200w, /_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=1920&q=100 1920w, /_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=2048&q=100 2048w, /_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=3840&q=100 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
    </div>
  </figure>
</DocumentFragment>
`;

exports[`FullWidthImage renders without alt text 1`] = `
<DocumentFragment>
  <figure>
    <div
      class="relative overflow-hidden mb-xs aspect-4/3"
    >
      <img
        alt=""
        class="object-cover"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        sizes="100vw"
        src="/_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=3840&q=100"
        srcset="/_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=640&q=100 640w, /_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=750&q=100 750w, /_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=828&q=100 828w, /_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=1080&q=100 1080w, /_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=1200&q=100 1200w, /_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=1920&q=100 1920w, /_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=2048&q=100 2048w, /_next/image?url=http%3A%2F%2Fwww.not-a-real-url-this-is-just-for-testing.com%2Fimage.png&w=3840&q=100 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
    </div>
    <figcaption
      class="text-body-small text-gray-700"
    >
      This is a test image caption
    </figcaption>
  </figure>
</DocumentFragment>
`;
